src/frontend/staff-portal/src/app/components/staff-workbench/update-request-info/update-request-info.component.ts

```diff
@@ -0,0 +1,155 @@
+  public onSubmit(): void {
+    // process accepts and rejects
+    this.disputeUpdateRequests.forEach(disputeUpdateRequest => {
+      if (disputeUpdateRequest.status === this.RequestUpdateStatus.Pending && disputeUpdateRequest.newStatus === this.RequestUpdateStatus.Accepted) {
+        this.busy = this.disputeService.acceptDisputeUpdateRequest(disputeUpdateRequest.disputantUpdateRequestId).subscribe({
+          next: response => {
+            disputeUpdateRequest.status = this.RequestUpdateStatus.Accepted;
+          },
+          error: err => { },
+          complete: () => { }
+        });
+      } else if (disputeUpdateRequest.status === this.RequestUpdateStatus.Pending && disputeUpdateRequest.newStatus === this.RequestUpdateStatus.Rejected) {
+        this.busy = this.disputeService.rejectDisputeUpdateRequest(disputeUpdateRequest.disputantUpdateRequestId).subscribe({
+          next: response => {
+            disputeUpdateRequest.status = this.RequestUpdateStatus.Rejected;
+          },
+          error: err => { },
+          complete: () => { }
+        });      }
+    })
+    this.getDispute();
+  }
+
+  // get dispute
+  getDispute(): void {
+    this.logger.log('UpdateRequestInfoComponent::getDispute');
+
+    this.busy = this.disputeService.getDispute(this.disputeInfo.disputeId).subscribe((response: Dispute) => {
+      this.logger.info(
+        'UpdateRequestInfoComponent::getDispute response',
+        response
+      );
+
+      this.initialDisputeValues = response;
+
+      // set violation date and time
+      let tempViolationDate = this.initialDisputeValues?.issuedTs?.split("T");
+      if (tempViolationDate) {
+        this.violationDate = tempViolationDate[0];
+        this.violationTime = tempViolationDate[1].split(":")[0] + ":" + tempViolationDate[1].split(":")[1];
+      }
+
+      if (this.retrieving === true) this.getDisputeUpdateRequests(this.initialDisputeValues.disputeId);
+
+    });
+  }
+
+  getDisputeUpdateRequests(disputeId: number) {
+    this.logger.log('UpdateRequestInfoComponent::getDisputeUpdateRequests');
+
+    this.busy = this.disputeService.getDisputeUpdateRequests(disputeId).subscribe((response) => {
+      this.retrieving = false;
+      this.logger.info(
+        'UpdateRequestInfoComponent::getDisputeUpdateRequests response',
+        response
+      );
+
+      this.disputeUpdateRequests = response;
+
+      // sort oldest to newest
+      this.disputeUpdateRequests = this.disputeUpdateRequests.sort((a,b) => {if (a.createdTs < b.createdTs) return -1});
+
+    });
+  }
+
+  // respond to status changes from child component
+  onDisputantUpdateRequestStatusChange(disputantUpdateRequest: DisputantUpdateRequest) {
+    this.disputeUpdateRequests.forEach(disputeUpdateRequest => {
+      if (disputeUpdateRequest.disputantUpdateRequestId === disputantUpdateRequest.disputantUpdateRequestId) disputeUpdateRequest.status = disputantUpdateRequest.status;
+    });
+  }
+
+  getFormattedAddress(): string {
```

Comment:
should we put this function inside dispute.service.ts ?

Reply:
if it gets used a 2nd time then yes we should move it there